### ISSUE TITLE
 Mathematica installation problems on a machine running ClearLinux

### ISSUE THREAD

**Issue by dhalpern1** [clearlinux/distribution#1325]
I am trying to install Mathematica on a machine that is running Clear Linux.
I get some errors that prevent me from installing the software.

(1) During the installation process I get the following ouput: (a) [CRITICAL ERROR: InstallSize not defined.                                                                                                                             ]
 
and (b) Installation failed. See /tmp/InstallErrors-3160.

(2) The content of the file mentioned in (1)(b) is:
zip: abort: internal threads error
expr: syntax error: unexpected argument '1024'
Critical Failure in CheckSpace_()
 

I've contacted the developers of Mathematica, Wolfram, and they are not too helpful. They are basically telling me that I am not using one of the approved Linux versions from Red  Hat or Ubuntu.

Any help would be appreciated about what may be going on.

Thanks.

David Halpern

**Comment by ahkok** [comment]
Since this is proprietary software that we do not have access to, there is very little that we can do to help.

**Comment by dhalpern1** [comment]
Your response is not at all helpful. I am fully that Mathematica is not an open piece of software. 

It seems to say that there could be a problem with zip? Surely that's open source?


> Since this is proprietary software that we do not have access to, there is very little that we can do to help.

**Comment by heterotic** [comment]
Hi,

I ran into a similar issue previously. The Mathematica 12.0 installer makes a call to ```gunzip -l```, i.e. ```gzip -d -l```, and this option combination confuses pigz. Additionally, at least on my system and Mathematica version, pigz can't calculate the decompressed size of the Mathematica archive reliably and appends a question mark character which the installer doesn't know how to parse. Since the installer checks ```/usr/contrib/bin``` before ```/usr/bin```, creating a ```/usr/contrib/bin/gunzip``` wrapper
```bash
#!/usr/bin/env bash
if [ "$1" == "-l" ]; then
    /usr/bin/gzip "$@" | sed 's/?//g'
else
    /usr/bin/gunzip "$@"
fi 
```
with appropriate permissions allowed the installation to continue if there was adequate disk space.

**Comment by bryteise** [comment]
@heterotic Oh that's good to know, thanks. Would adding a ~/bin/gunzip in path before /usr/bin also work or is the installer script hard coding things?

**Comment by heterotic** [comment]
@bryteise From what I can tell without violating the license agreement, it's hard coded.

**Comment by hhcorzo** [comment]
Hello, I just installed Clear Linux but I have the same problem when I try to install Mathematica 12. Have been any solutions for this issue?

**Comment by bryteise** [comment]
I think that the [workaround](https://github.com/clearlinux/distribution/issues/1325#issuecomment-541274993) heterotic posted is probably the easiest thing I can suggest.

**Comment by hhcorzo** [comment]
I am new in linux,  so, the only thing I need to do is to make a file with:

```
#!/usr/bin/env bash
if [ "$1" == "-l" ]; then
    /usr/bin/gzip "$@" | sed 's/?//g'
else
    /usr/bin/gunzip "$@"
fi 
```
and run it before my mathematica installation?

**Comment by lebensterben** [comment]
>I am new in linux,  so, the only thing I need to do is to make a file with: ...

@hhcorzo Not exactly. It says to create a file `/use/contrib/bin/gunzip` with these contents. And the installer would source this file.

**Comment by hhcorzo** [comment]
Oh! I see it worked. Thanks!
